### PR TITLE
Inu 264 8 dress keychain wrapper

### PR DESCRIPTION
**Pull request by SanakaBarabaka**
All files was copied from private repo.
Only changes are:
- ./README.md
- ./Keychain/README.md

### CODE REVIEW

File: Keychain/README.md [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+
+## Usage
+
+Import module:
+
+```swift
+import Shakuro_iOS_Toolbox
+```
+
+KeychainWrapper work with `Codable` objects:
```
Comment (vonipchenko) [inline Keychain/README.md]: WORKS

File: Keychain/README.md [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+    static let keychainServiceName: String = "com.shakuro.testApp"
+    static let keychainItemId: String = "testUserCredentials"
+}
+```
+
+Save data into Keychain:
+
+```swift
+func saveItemToKeychain() {
+    let credentials = UserCredentials(username: "Vasya", password: "123")
```
Comment (vonipchenko) [inline Keychain/README.md]: Please replace "Vasya" with myUsername and  "123" with myPassword

File: Keychain/Source/KeychainWrapper.swift [hunk 3]
```diff
@@ -0,0 +1,181 @@
...
+    }
+
+    public struct Item<T> {
+        public let serviceName: String
+        public let account: String
+        public let itemName: String?
+        public let accessGroup: String?
+        public let secValue: T
+
+        public init(serviceName: String, account: String, itemName: String? = nil, accessGroup: String? = nil, secValue: T) {
```
Comment (vonipchenko) [inline Keychain/Source/KeychainWrapper.swift]: "Place parameters that don’t have default values at the beginning of a function’s parameter list, before the parameters that have default values. Parameters that don’t have default values are usually more important to the function’s meaning—writing them first makes it easier to recognize that the same function is being called, regardless of whether any default parameters are omitted."

File: Keychain/Source/KeychainWrapper.swift [hunk 4]
```diff
@@ -0,0 +1,181 @@
...
+        }
+
+    }
+
+    // MARK: - Public
+
+    /**
+     - throws: KeychainWrapper.Error
+     */
+    public static func saveKeychainItem<T>(_ item: KeychainWrapper.Item<T>) throws where T:Encodable {
```
Comment (vonipchenko) [inline Keychain/Source/KeychainWrapper.swift]: missing whitespace T:Encodable
Please check via linter

File: Keychain/Source/KeychainWrapper.swift [hunk 5]
```diff
@@ -0,0 +1,181 @@
...
+        case unexpectedKeychainItemData(queryResult: AnyObject?)
+        case encodeSecValueError
+        case addKeychainItemError(osStatus: OSStatus)
+        case readKeychainItemError(osStatus: OSStatus)
+        case updateKeychainItemError(osStatus: OSStatus)
+        case deleteKeychainItemError(osStatus: OSStatus)
+        case searchKeychainError(osStatus: OSStatus)
+    }
+
+    public struct Item<T> {
```
Comment (vonipchenko) [inline Keychain/Source/KeychainWrapper.swift]: missing generic constraint to Encodable

File: Keychain/Source/KeychainWrapper.swift [hunk 6]
```diff
@@ -0,0 +1,181 @@
...
+                }
+                status = SecItemUpdate(updateQuery as CFDictionary, attributesToUpdate as CFDictionary)
+                if status != noErr {
+                    throw KeychainWrapper.Error.updateKeychainItemError(osStatus: status)
+                }
+            default:
+                throw KeychainWrapper.Error.searchKeychainError(osStatus: status)
+            }
+        } else {
+            throw KeychainWrapper.Error.encodeSecValueError
```
Comment (vonipchenko) [inline Keychain/Source/KeychainWrapper.swift:81]: Please do not hide JSONEncoder() original error.
I think there two options:
1) Just use try JSONEncoder().encode(item.secValue) to leave original error
2) Add associated value to the "encodeSecValueError" with original error from JSONEncoder

**Review by vonipchenko — CHANGES_REQUESTED** [review]

**Comment by SanakaBarabaka** [comment]
add ability to get transform from preview layer of video camera - this is used with metadata output for correct frame transformation

File: Keychain/Source/KeychainWrapper.swift (hunk 6 above)
Comment (SanakaBarabaka) [reply Keychain/Source/KeychainWrapper.swift:81]: added TODO to private repo

**Pull request merged**